### DISCUSSION TITLE
Forking Chains (interesting question)

### DISCUSSION THREAD

**Discussion by d4r3-w0lf** [Cyfrin/foundry-full-course-cu#2334] [General]
When we use this command to fork a blockchain:

forge test -mt testPriceFeedVersionIsAccurate --fork-url $SEPOLIA_RPC_URL

We might wonder if it copies all the data from the chosen network. Big blockchains like Ethereum, Polygon, and Solana have a lot of data, so copying everything should take a long time. But when we run this command, it finishes quickly. This is surprising. How can it work so fast? We need to understand what's really happening when we fork a blockchain this way.

**Comment by EngrPips** [comment]
When you run the command, it only copies the data you need to run your process from the blockchain you are forking and not all the blockchain data. I believe Patrick mentioned this in the course.
